### PR TITLE
Extend Katib GH action to also create an experiment and assert it succeeds

### PR DESCRIPTION
**Pull request by NickLoukas**
Refs #2248

### CODE REVIEW

**Comment by kimwnasptd** [comment]
cc @kubeflow/wg-automl-leads the goal is to just sanity check that Experiments can work as expected alongside the common dependencies of Kubeflow. 

In the future once we'll move to AWS we'll just run the E2E tests for these, but for now we'll be testing each component separately

File: tests/gh-actions/katib_test.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+---
```
Comment (kimwnasptd) [inline tests/gh-actions/katib_test.yaml:1]: only nit I have would be to put the files/yamls we'll need for these tests into a distinct folder and not have everything flat under `tests/gh-actions`. Let's do `tests/gh-actions/kf-objects` WDYT?

Comment (NickLoukas) [reply tests/gh-actions/katib_test.yaml:1]: Yeap, I agree. I'll add a commit to arrange the folders like that.

**Comment by NickLoukas** [comment]
@kimwnasptd I added an extra commit here for the review comment above.

**Comment by kimwnasptd** [comment]
Thanks! 

/lgtm
/approve

**Pull request merged**